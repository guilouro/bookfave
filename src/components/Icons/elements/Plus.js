import React from 'react'

const Plus = props => (
  <svg viewBox="0 0 26 26" xmlns="http://www.w3.org/2000/svg" {...props}>
    <path
      d="M18.121 11.72h-3.84V7.879a1.28 1.28 0 0 0-2.561 0v3.84H7.879a1.28 1.28 0 0 0 0 2.561h3.84v3.841a1.28 1.28 0 0 0 2.561 0v-3.84h3.841a1.28 1.28 0 0 0 0-2.561zm5.121 7.682a3.845 3.845 0 0 1-3.84 3.84H6.598a3.845 3.845 0 0 1-3.84-3.84V6.598a3.845 3.845 0 0 1 3.84-3.84h12.804a3.845 3.845 0 0 1 3.84 3.84v12.804zM19.402.197H6.598A6.409 6.409 0 0 0 .197 6.598v12.804a6.409 6.409 0 0 0 6.401 6.401h12.804a6.409 6.409 0 0 0 6.401-6.401V6.598A6.409 6.409 0 0 0 19.402.197z"
      fill="currentColor"
    />
  </svg>
)

export default Plus
