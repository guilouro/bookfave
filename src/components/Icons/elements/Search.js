import React from 'react'

const Search = props => (
  <svg viewBox="0 0 26 26" xmlns="http://www.w3.org/2000/svg" {...props}>
    <path
      d="M11.583 20.083c-4.687 0-8.5-3.812-8.5-8.5 0-4.687 3.813-8.5 8.5-8.5 4.688 0 8.5 3.813 8.5 8.5 0 4.688-3.812 8.5-8.5 8.5m13.752 3.249l-4.81-4.811a11.265 11.265 0 0 0 2.391-6.938C22.916 5.334 17.832.25 11.583.25 5.334.25.25 5.334.25 11.583c0 6.25 5.084 11.334 11.333 11.334 2.615 0 5.02-.9 6.937-2.392l4.811 4.81a1.413 1.413 0 0 0 2.004 0 1.415 1.415 0 0 0 0-2.003"
      fill="currentColor"
    />
  </svg>
)

export default Search
