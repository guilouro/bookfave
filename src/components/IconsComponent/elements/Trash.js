import React from 'react'

const Trash = props => (
  <svg viewBox="0 0 16 18" xmlns="http://www.w3.org/2000/svg" {...props}>
    <path
      d="M12 15c0 .551-.448 1-1 1H5c-.552 0-1-.449-1-1V6h8v9zM8 2c1.103 0 2 .897 2 2H6c0-1.103.897-2 2-2zm7 2h-3c0-2.206-1.794-4-4-4S4 1.794 4 4H1a1 1 0 1 0 0 2h1v9c0 1.654 1.346 3 3 3h6c1.654 0 3-1.346 3-3V6h1a1 1 0 1 0 0-2z"
      fill="currentColor"
    />
  </svg>
)

export default Trash
