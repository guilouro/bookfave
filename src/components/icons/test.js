import React from 'react'
import { render } from 'react-testing-library'
import Icon from '.'
import * as AllIcons from './elements'

describe('<Icon />', () => {
  Object.keys(AllIcons).forEach(CurrentIcon => {
    it(`Should render <${CurrentIcon} /> correctly`, () => {
      const { container } = render(<Icon name={CurrentIcon} />)
      expect(container).toMatchSnapshot()
    })
  })
})
