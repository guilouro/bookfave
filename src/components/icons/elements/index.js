import Trash from './Trash'
import Plus from './Plus'
import Search from './Search'
import Empty from './Empty'

export { Trash, Plus, Search, Empty }
