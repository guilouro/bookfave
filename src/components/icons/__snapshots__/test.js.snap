// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> Should render <Empty /> correctly 1`] = `
<div>
  <svg
    fill="currentColor"
    height="20px"
    viewBox="0 0 475 475.884"
    width="20px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M473.738 328.543A23.656 23.656 0 0 0 450.922 312h-10.48V56a8 8 0 0 0-8-8h-112V8a8 8 0 0 0-8-8h-304a8 8 0 0 0-8 8v416a8 8 0 0 0 8 8h264.32l-4 12.336a23.993 23.993 0 0 0 8.695 26.879 24 24 0 0 0 28.25-.016l41.934-30.543a8.06 8.06 0 0 1 9.511 0l42.012 30.559a23.655 23.655 0 0 0 28.223 0 23.66 23.66 0 0 0 8.718-26.871l-16.062-49.489a7.915 7.915 0 0 1 2.934-8.863l42.09-30.558a23.71 23.71 0 0 0 8.671-26.891zM424.441 64v248h-25.52a7.941 7.941 0 0 1-2.472-.406 8.15 8.15 0 0 1-5.117-5.106L375.25 257.04a24.002 24.002 0 0 0-45.602 0l-9.207 28.266V208h64a8 8 0 0 0 0-16h-64v-32h64a8 8 0 0 0 0-16h-64v-32h64a8 8 0 0 0 0-16h-64V64zm-408 352V16h288v296h-50.48a24.004 24.004 0 0 0-22.879 16.566 24.01 24.01 0 0 0 8.734 26.868l42.075 30.55a7.927 7.927 0 0 1 2.945 8.903L277.969 416zm439.2-73.504l-42.079 30.559a23.87 23.87 0 0 0-8.738 26.777l16.082 49.441a7.999 7.999 0 0 1-12.32 8.973l-42.04-30.582a24.2 24.2 0 0 0-28.265 0l-41.992 30.559a7.999 7.999 0 1 1-12.328-8.965l16.074-49.418a23.883 23.883 0 0 0-8.746-26.817l-42.047-30.52A8.002 8.002 0 0 1 253.961 328h52a22.367 22.367 0 0 0 9.648-2.098l.153-.086.144-.046c.114-.051.192-.145.305-.204a23.754 23.754 0 0 0 12.559-14.132L344.852 262a8.003 8.003 0 0 1 7.597-5.504 8.004 8.004 0 0 1 7.602 5.504l16.086 49.473A24.006 24.006 0 0 0 398.922 328h52a7.995 7.995 0 0 1 7.664 5.52 7.994 7.994 0 0 1-2.938 8.976zm0 0"
    />
    <path
      d="M72.441 80h176a8 8 0 0 0 0-16h-176a8 8 0 0 0 0 16zm0 0M72.441 144h176a8 8 0 0 0 0-16h-176a8 8 0 0 0 0 16zm0 0M72.441 208h176a8 8 0 0 0 0-16h-176a8 8 0 0 0 0 16zm0 0M256.441 264a8 8 0 0 0-8-8h-176a8 8 0 0 0 0 16h176a8 8 0 0 0 8-8zm0 0"
    />
  </svg>
</div>
`;

exports[`<Icon /> Should render <Plus /> correctly 1`] = `
<div>
  <svg
    height="20px"
    viewBox="0 0 26 26"
    width="20px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M18.121 11.72h-3.84V7.879a1.28 1.28 0 0 0-2.561 0v3.84H7.879a1.28 1.28 0 0 0 0 2.561h3.84v3.841a1.28 1.28 0 0 0 2.561 0v-3.84h3.841a1.28 1.28 0 0 0 0-2.561zm5.121 7.682a3.845 3.845 0 0 1-3.84 3.84H6.598a3.845 3.845 0 0 1-3.84-3.84V6.598a3.845 3.845 0 0 1 3.84-3.84h12.804a3.845 3.845 0 0 1 3.84 3.84v12.804zM19.402.197H6.598A6.409 6.409 0 0 0 .197 6.598v12.804a6.409 6.409 0 0 0 6.401 6.401h12.804a6.409 6.409 0 0 0 6.401-6.401V6.598A6.409 6.409 0 0 0 19.402.197z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<Icon /> Should render <Search /> correctly 1`] = `
<div>
  <svg
    height="20px"
    viewBox="0 0 26 26"
    width="20px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M11.583 20.083c-4.687 0-8.5-3.812-8.5-8.5 0-4.687 3.813-8.5 8.5-8.5 4.688 0 8.5 3.813 8.5 8.5 0 4.688-3.812 8.5-8.5 8.5m13.752 3.249l-4.81-4.811a11.265 11.265 0 0 0 2.391-6.938C22.916 5.334 17.832.25 11.583.25 5.334.25.25 5.334.25 11.583c0 6.25 5.084 11.334 11.333 11.334 2.615 0 5.02-.9 6.937-2.392l4.811 4.81a1.413 1.413 0 0 0 2.004 0 1.415 1.415 0 0 0 0-2.003"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<Icon /> Should render <Trash /> correctly 1`] = `
<div>
  <svg
    height="20px"
    viewBox="0 0 16 18"
    width="20px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 15c0 .551-.448 1-1 1H5c-.552 0-1-.449-1-1V6h8v9zM8 2c1.103 0 2 .897 2 2H6c0-1.103.897-2 2-2zm7 2h-3c0-2.206-1.794-4-4-4S4 1.794 4 4H1a1 1 0 1 0 0 2h1v9c0 1.654 1.346 3 3 3h6c1.654 0 3-1.346 3-3V6h1a1 1 0 1 0 0-2z"
      fill="currentColor"
    />
  </svg>
</div>
`;
